internal/impl/packages.go

```diff
@@ -40,8 +40,19 @@ func (d *Devbox) Add(ctx context.Context, pkgsNames ...string) error {
 
 	// Only add packages that are not already in config. If same canonical exists,
 	// replace it.
-	pkgs := []*devpkg.Package{}
-	for _, pkg := range devpkg.PackageFromStrings(lo.Uniq(pkgsNames), d.lockfile) {
+	pkgs := devpkg.PackageFromStrings(lo.Uniq(pkgsNames), d.lockfile)
+	for _, pkg := range pkgs {
+		// Resolving here ensures we allow insecure before running ensurePackagesAreInstalled
+		// which will call print-dev-env. Resolving does not save the lockfile, we
+		// save at the end when everything has succeeded.
+		p, err := d.lockfile.Resolve(pkg.Raw)
+		if err != nil {
+			return err
```

Comment:
Will this error for the fallthrough case?

Reply:
good catch. Will test and fix.

Reply:
This works, because in the fallthrough case resolve works fine. It actually enters non-fallthrough packages twice (and then cleans it up when `lock.Tidy()` is called). I'm gonna see if I can make this better.